server/pulp/plugins/util/publish_step.py

```diff
@@ -858,7 +858,7 @@ def process_main(self):
         """
         if self.delete_before_copy:
             shutil.rmtree(self.target_dir, ignore_errors=True)
-        shutil.copytree(self.source_dir, self.target_dir, symlinks=self.preserve_symlinks)
+        os.system("cp -a %s %s" % (self.source_dir, timestamp_master_dir))
```

Comment:
This will break export and iso export codepaths because the resulting publish would by all symlinks which will break when they are disconnected from the base filesystem with the actual content. To fix this have the `self.preserve_symlinks` option pass the right options to `cp` to either preserve or dereference symlinks.